wallet/tutorials/javascript-dapp-simple.md

```diff
@@ -0,0 +1,317 @@
+ You can install the [Live Server](https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer)
+extension for VS Code to easily launch a local development server for your dapp.
+
+- [npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm).
+
+## Steps
+
+### 1. Set up a new project
+
+Create a new project with the following structure:
+
+```text
+simple-dapp/
+├─ src/
+│  ├─ index.js
+├─ dist/
+│  ├─ index.html
+```
+This tutorial will use the [`@metamask/detect-provider`](https://github.com/MetaMask/detect-provider) module and a bundler, [Webpack](https://github.com/webpack/webpack), to compile the module and create an output script `dist/main.js`.
+
+Add the following code to `dist/index.html`:
+
+```jsx title="index.html"
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Simple dapp</title>
+</head>
+<body>
+    <script src="../dist/index.js"></script>  <!-- Links to the JavaScript bundle created by Webpack -->
+</body>
+</html>
+```
+
+### 2. Install and configure Webpack
+
+Navigate to the root of your `simple-dapp` directory and run the following commands to initialize npm and install Webpack.
+
+```bash 
+npm init -y && npm install --save-dev webpack webpack-cli
+```
+
+Create a `webpack.config.js` in the root of your project with the following configuration:
+
+```js title="webpack.config.js"
+const path = require("path");
+
+module.exports = {
+  entry: "./src/index.js",
+  output: {
+    path: path.resolve(__dirname, "dist"),
+    filename: "index.js",
+  },
+  mode: "development",
+};
+```
+
+Then, edit `package.json` to include a build script:
+
+```json title="package.json"
+"scripts": {
+  "build": "webpack"
+}
+```
+
+When you run `npm run build` from your terminal, Webpack bundles the content in `src/index.js` into `dist/main.js`, which is linked in `index.html`.
+
+### 3. Detect MetaMask
+
+:::caution
+
+Using the `window.ethereum` provider object to detect MetaMask is deprecated. We recommend [connecting to MetaMask using EIP-6963](../how-to/connect/index.md) for production environments.
+
+:::
+
+Install the `@metamask/detect-provider` module in your project directory:
+
+```bash
+npm i @metamask/detect-provider
+```
+
+Add the following code to `src/index.js` to detect the MetaMask provider using `@metamask/detect-provider`:
+
+```js title="index.js"
+import detectEthereumProvider from "@metamask/detect-provider";
+
+async function setup() {
+  const provider = await detectEthereumProvider();
+
+  if (provider && provider === window.ethereum) {
+    console.log("MetaMask is available!");
+    startApp(provider); // Initialize your dapp with MetaMask.
+  } else {
+    console.log("Please install MetaMask!");
+  }
+}
+
+function startApp(provider) {
+  if (provider !== window.ethereum) {
+    console.error("Do you have multiple wallets installed?");
+  }
+}
+
+window.addEventListener("load", setup);
+```
+
+#### Serve the project
```

Comment:
```suggestion
### 4. Serve the project
```
Add as separate step?

Reply:
I think this part of detect Metamask. I am not sure this needs to be a separate step.  I think it is more of a sub-step of what is described earlier

Reply:
This step is testing the project in the local server, which is a separate task to installing and using the `detect-provider` package. This step is actual useful to use again throughout the rest of the tutorial, and at the end for testing the finished product.